src/main/java/com/metamx/emitter/core/ParametrizedUriEmitter.java

```diff
@@ -120,10 +121,30 @@ public void close() throws IOException
   @Override
   public void flush() throws IOException
   {
-    synchronized (emitters) {
-      for (Emitter emitter : emitters.values()) {
-        emitter.flush();
+    Exception thrown = null;
+    for (HttpPostEmitter httpPostEmitter : emitters.values()) {
+      try {
+        httpPostEmitter.flush();
       }
+      catch (Exception e) {
+        // If flush was interrupted, exit the loop
+        if (Thread.currentThread().isInterrupted()) {
```

Comment:
Specifically in `HttpPostEmitter.flush()` the flag is reset. I think the convention should be that either you throw `InterruptedException` (*not* wrapped), or the `Thread`'s interrupted flag is set. This policy is defined in Java Concurrency in Practice: https://books.google.ru/books?id=EK43StEVfJIC&lpg=PA142&ots=upTyBauTmw&dq=jcip%20interrupted&pg=PA142#v=onepage&q&f=false

If this policy is violated, it's a defect in the code that violates, and we shouldn't always defensively protect from that. `HttpPostEmitter.flush()` adheres it.